manifests/charts/istio-control/istio-discovery/templates/configmap.yaml

```diff
@@ -5,6 +5,11 @@
     trustDomain: {{ .Values.global.trustDomain | quote }}
 
     defaultConfig:
+      {{- if .Values.global.meshID }}
```

Comment:
I would recommend:
* Deprecate this option (https://github.com/istio/istio/blob/07782fd8d2ff0225af494b172dc459e5c9ace04a/operator/pkg/apis/istio/v1alpha1/validation/validation.go#L81) in favor of directly configuring mesh config
* Start reading this in the code. Setting this here but not using it is bound to cause confusion

Reply:
We put `meshID` in the sidecar injection template, but it isn't accessible anywhere else. This is a step towards replacing that bit of functionality -- needed for telemetry -- (and making it available for inspection where needed).

We can deprecate the `values.global.meshID` in favor of direct meshconfig setting, but that seems like a separate concern, and one that should be tackled at the same time as the sidecar injection template update.

Reply:
My understanding is today meshID -> ISTIO_META_MESH_ID env var in injector and gateway -> node metadata. We have been moving to move logic into the agent instead of the injector/gateway charts to minimize our dependency on helm,  simplify the deployments, and have a single source of truth (mesh config). what I would envision this is like now is meshID is in proxy config -> pilot-agent gets proxy config -> pilot-agent sets the node metadata based on the proxy config

Reply:
If we move this all into agent code, that's fine. When will the pilot-agent proxy-config node metadata stuff be ready to go?